### ISSUE TITLE
Entity Framework - Entity with List

### ISSUE THREAD

**Issue by M2xAy** [ChilliCream/hotchocolate#4887]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hi Community

I receive following errors if I have a List Object in my Entity that I added as QueryType
but if I add GraphQlIgnore that works
  [GraphQLIgnore]
  public List<Analyt.Analyt> Analyten {get;set;}

1. Unable to infer or resolve a schema type from the type reference `Input: IDbContextTransaction`. (HotChocolate.Types.InputObjectType<Microsoft.EntityFrameworkCore.Infrastructure.DatabaseFacade>)
2. Unable to infer or resolve a schema type from the type reference `Input: IModel`.
3. Unable to infer or resolve a schema type from the type reference `Input: DbContextId`.

I have following Nugets:
```
	  <PackageReference Include="HotChocolate.AspNetCore" Version="12.7.0" />
	  <PackageReference Include="HotChocolate.AspNetCore.Authorization" Version="12.7.0" />
	  <PackageReference Include="HotChocolate.Data.EntityFramework" Version="12.7.0" />
	  <PackageReference Include="HotChocolate.Types.OffsetPagination" Version="12.7.0" />
```

Can you help me?
Best regards from Zurich

### Steps to reproduce

1.     [GraphQLIgnore]
      public List<Analyt.Analyt> Analyten {get;set;}
2.      public List<Analyt.Analyt> Analyten {get;set;}

### Relevant log output

```shell
HotChocolate.SchemaException: For more details look at the `Errors` property.

1. Unable to infer or resolve a schema type from the type reference `Input: IDbContextTransaction`. (HotChocolate.Types.InputObjectType<Microsoft.EntityFrameworkCore.Infrastructure.DatabaseFacade>)
2. Unable to infer or resolve a schema type from the type reference `Input: IModel`.
3. Unable to infer or resolve a schema type from the type reference `Input: DbContextId`.

   at HotChocolate.Configuration.TypeInitializer.DiscoverTypes()
   at HotChocolate.Configuration.TypeInitializer.Initialize()
   at HotChocolate.SchemaBuilder.Setup.InitializeTypes(SchemaBuilder builder, IDescriptorContext context, IReadOnlyList`1 types, LazySchema lazySchema)
   at HotChocolate.SchemaBuilder.Setup.Create(SchemaBuilder builder, LazySchema lazySchema, IDescriptorContext context)
   at HotChocolate.SchemaBuilder.Create(IDescriptorContext context)
   at HotChocolate.SchemaBuilder.HotChocolate.ISchemaBuilder.Create(IDescriptorContext context)
   at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaAsync(NameString schemaName, RequestExecutorSetup options, RequestExecutorOptions executorOptions, IServiceProvider serviceProvider, TypeModuleChangeMonitor typeModuleChangeMonitor, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaServicesAsync(NameString schemaName, RequestExecutorSetup options, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorNoLockAsync(NameString schemaName, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorAsync(NameString schemaName, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorProxy.GetRequestExecutorAsync(CancellationToken cancellationToken)
   at HotChocolate.AspNetCore.HttpPostMiddlewareBase.HandleRequestAsync(HttpContext context, AllowedContentType contentType)
   at HotChocolate.AspNetCore.HttpPostMiddlewareBase.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.<>c__DisplayClass13_0.<<UseCancellation>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished HTTP/2 POST https://localhost:44326/graphql application/json 266 - 500 - text/plain;+charset=utf-8 1431.1810ms
```

### Additional Context?

-

### Product

Hot Chocolate

### Version

12.7.0

**Comment by matt-psaltis** [comment]
It seems strange that the InputObjectType would have the DatabaseFacade as the generic type parameter. That should typically be the model class. Its difficult to tell from the information provided where that fault may come from.

**Comment by M2xAy** [comment]
Thank you for your answer.
If you want, I can create a test Project
Best regards

**Comment by michaelstaib** [comment]
It looks like you forgot to add the `[ServiceAttribute]` which will lead to the schema engine inferring an input from your dbcontext.

**Comment by michaelstaib** [comment]
Please provide us with a repro or some more code otherwise it looks like what I have stated above.

**Comment by M2xAy** [comment]
@michaelstaib  Thank you for your Suggestion. After your comment, I searched more and find the solution.
IAdded this and now works. I dont know why ;) I hope this will also help to other
   .RegisterDbContext<AppContext>()

**Comment by M2xAy** [comment]
> @M2xAy did you get this information from the [docs](https://chillicream.com/docs/hotchocolate/integrations/entity-framework/#registerdbcontext)? If so, could you let me know why its still unclear and what's confusing to you?

Hi tobias
i first tested graphql with another simple project, everything worked fine, then i installed and configured the same setup in another big complex project and i had the error with the lists in dbentity. so i am confused because RegisterDbContext was not in the first project.

After Michael's comment, I googled it and found the documentation you linked. I tested it and it works.

Why I am confused: because without RegisterDbContext it worked in the first simple project.

With kind regards
Murat
